cql3/restrictions/statement_restrictions.cc

```diff
@@ -1012,8 +1012,18 @@ dht::partition_range_vector partition_ranges_from_token(const expr::expression&
     return {{std::move(start), std::move(end)}};
 }
 
+void validate_key_length(const std::vector<managed_bytes>& key, const size_t max_length) {
+    for (auto k : key) {
+        if (k.size() > max_length) {
+            throw exceptions::invalid_request_exception(format("Key length of {} is longer than maximum of {}", k.size(), max_length));
+        }
+    }
+}
+
 /// Turns a partition-key value into a partition_range. \p pk must have elements for all partition columns.
 dht::partition_range range_from_bytes(const schema& schema, const std::vector<managed_bytes>& pk) {
+    // partition key length can't be  greater then 65535
+    validate_key_length(pk, std::numeric_limits<uint16_t>::max());
```

Comment:
Nitpick: in other places below instead of using uint16_t, you used some alias like compound_type::size_type. Do we not have such an alias here?
And if we don't, does it really make sense for the size type to be an option for validate_key_length - shouldn't it enforce the same uint16_t limit everywhere it's called?

Reply:
Is it possible to have different restrictions for different types of keys?
I've been a little bit confused with cassandra behavior.

I guess it's better to keep two parameters in the function but have no idea which alias is better to use.

Reply:
@nyh ok, I've changed it according to your suggestions